nexus/db-queries/src/db/datastore/ereport.rs

```diff
@@ -0,0 +1,423 @@
+        slot: SpMgsSlot,
+    ) -> impl RunnableQuery<EreportIdTuple> {
+        sp_dsl::sp_ereport
+            .filter(sp_dsl::sp_type.eq(sp_type).and(sp_dsl::sp_slot.eq(slot)))
+            .order_by((sp_dsl::time_collected.desc(), sp_dsl::ena.desc()))
+            .limit(1)
+            .select((sp_dsl::restart_id, sp_dsl::ena))
+    }
+
+    pub async fn host_latest_ereport_id(
+        &self,
+        opctx: &OpContext,
+        sled_id: SledUuid,
+    ) -> Result<Option<ereport_types::EreportId>, Error> {
+        opctx.authorize(authz::Action::ListChildren, &authz::FLEET).await?;
+        let id = Self::host_latest_ereport_id_query(sled_id)
+            .get_result_async(&*self.pool_connection_authorized(opctx).await?)
+            .await
+            .optional()
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?
+            .map(id_from_tuple);
+        Ok(id)
+    }
+
+    fn host_latest_ereport_id_query(
+        sled_id: SledUuid,
+    ) -> impl RunnableQuery<EreportIdTuple> {
+        host_dsl::host_ereport
+            .filter(host_dsl::sled_id.eq(sled_id.into_untyped_uuid()))
+            .order_by((host_dsl::time_collected.desc(), host_dsl::ena.desc()))
+            .limit(1)
+            .select((host_dsl::restart_id, host_dsl::ena))
+    }
+
+    pub async fn sp_ereports_insert(
+        &self,
+        opctx: &OpContext,
+        sp_type: impl Into<SpType>,
+        slot: u16,
+        ereports: Vec<SpEreport>,
+    ) -> CreateResult<(usize, Option<ereport_types::EreportId>)> {
+        opctx.authorize(authz::Action::CreateChild, &authz::FLEET).await?;
+        let conn = self.pool_connection_authorized(opctx).await?;
+        let created = diesel::insert_into(sp_dsl::sp_ereport)
+            .values(ereports)
+            .on_conflict((sp_dsl::restart_id, sp_dsl::ena))
+            .do_nothing()
+            .execute_async(&*conn)
+            .await
+            .map_err(|e| {
+                public_error_from_diesel(e, ErrorHandler::Server)
+                    .internal_context("failed to insert ereports")
+            })?;
+        let sp_type = sp_type.into();
+        let latest = self
+            .sp_latest_ereport_id_on_conn(&conn, sp_type, slot)
+            .await
+            .map_err(|e| {
+                e.internal_context(format!(
+                    "failed to refresh latest ereport ID for {sp_type:?} {slot}"
+                ))
+            })?;
+        Ok((created, latest))
+    }
+}
+
+fn id_from_tuple(
+    (restart_id, DbEna(ena)): EreportIdTuple,
+) -> ereport_types::EreportId {
+    ereport_types::EreportId {
+        restart_id: EreporterRestartUuid::from_untyped_uuid(restart_id),
+        ena,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::db::explain::ExplainableAsync;
+    use crate::db::pub_test_utils::TestDatabase;
+    use crate::db::raw_query_builder::expectorate_query_contents;
+    use omicron_test_utils::dev;
+
+    #[tokio::test]
```

Comment:
These tests suggest to me that the "{sp,host}_latest_ereport_id" queries used to be more complicated/manual than they are now. Are these tests still providing value, or are they just change-detectors?